plugins/primus_lisp/lisp/simple-memory-allocator.lisp

```diff
@@ -43,8 +43,18 @@
         (if failed 0
           (set brk (+ brk n))
           (malloc/fill-edges ptr n)
-          (+ ptr *malloc-guard-edges*))))))
-
+          (let ((ptr (+ ptr *malloc-guard-edges*)))
```

Comment:
Dictionaries are not a part of the program state. We use dictionaries as storage for various analyses nothing more. Therefore, the semantics of a program shall depend only of the program state, which is fully described by `Env` and `Memory` components.